sdl2promela/sdl/model.py

```diff
@@ -1095,7 +1112,20 @@ def convert(source: ogAST.ProcedureCall):
 def convert(source: ogAST.Terminator) -> Action:
     if source.kind == "next_state":
         if source.inputString == "-":
-            return None  # No state switch
+            if any(next_id for next_id in source.candidate_id.keys() if next_id != -1):
```

Comment:
This function is growing.... But I guess not splitting it makes it more readable, so let's leave it as it.
However, it took some some time to figure this out. Why can we do:

for transition_id, states in source.candidate_id.items():

...

And then check if the len(result.candidates) > 0?

This could be shorter, with less nesting and I think the logic would be a bit simpler. What do you think?

Reply:
Ok, thish should work